tools/SetupFlow/DevHome.SetupFlow/ViewModels/DevDriveViewModel.cs

```diff
@@ -241,7 +238,7 @@ public DevDriveTaskGroup TaskGroup
     /// Opens folder picker and adds folder to the drive location, if the user does not cancel the dialog.
     /// </summary>
     [RelayCommand]
-    public async void ChooseFolderLocation()
+    public async Task ChooseFolderLocation()

```

Comment:
Yes, if the method returns a `Task` type, the MVVM Toolkit generator will wrap the method in an `AsyncRelayCommand` type (rather than `RelayCommand`). This by default has the following characteristics (when no other options are set):
- The method invoked by XAML will still await the task on the calling context (UI thread), so if an exception is thrown, the app will still crash in the same way, so there's no functional difference there.
- Concurrent executions are disabled, so the command will also report its "CanExecute" status changing to disabled while it's running. This will automatically eg. disable buttons when an asynchronous operation is running. In practice, this also means users can no longer eg. spam a button and start 10000 parallel executions of the same thing.
- It also exposes a handy "IsRunning" property, so you can eg. bind a progress bar or something to it very easily. That is, you no longer need to manually have some "isBusy" field to keep in sync when the method starts and completes.

You can also inspect the generated code yourself, of course.
The source for the command is [here](https://github.com/CommunityToolkit/dotnet/blob/main/src/CommunityToolkit.Mvvm/Input/AsyncRelayCommand.cs).

There's also a bunch of additional cool stuff you can enable, like, you can have a method also take a `CancellationToken` parameter, set `IncludeCancelCommand = true` on the attribute, and the generator will also automatically create and wire up a separate command to cancel the execution to this method, all properly raising the correct events for the UI as well.

Hope this helps! 🙂

Reply:
Love it 😍! Thanks for the informative details🙂
*(bookmarking)*